validations/src/validations.rs

```diff
@@ -724,7 +724,7 @@ pub fn validate_block(
     utxo_set: &UnspentOutputsPool,
     data_request_pool: &DataRequestPool,
     vrf: &mut VrfCtx,
-    rep_eng: &ReputationEngine,
```

Comment:
The `validate_block` function should not mutate anything, we need to find another place to update the reputation engine.

Reply:
The other option could be to put poe validation out of validate_block function, and update reputation engine before to call it

Reply:
(1/2) But why is this update needed? Isn't enough to update the reputation engine...